### PR TITLE
Added null safe operators and documentation for the .NET library

### PR DESCRIPTION
**Pull request by mackeyaj**
# PR Details
Adds the null safe operator to config variables and adds a readme file.

## Description
Allows the dotnet library to be imported without specifying resultDir or outDir vars in the pipeline_config. Adds readme file.

## Types of Changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I am submitting this pull request to the appropriate branch
- [x] I have labeled this pull request appropriately
- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: libraries/dotnet/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+---
+description: This library allows you to perform .Net build and test commands in the sdp dotnet-sdk:5.0 agent container
+---
+
+# Dotnet
```
Comment (ConnorDY) [inline libraries/dotnet/README.md:5]: I think we should write this as "DotNet".

File: libraries/dotnet/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+This library allows you to perform .Net build and test commands in the sdp dotnet-sdk:5.0 agent container.
```
Comment (ConnorDY) [inline libraries/dotnet/README.md:7]: ".Net" --> ".NET"

"sdp" --> "SDP"

Also, please add backticks (\`) around `dotnet-sdk:5.0` to make this styled a tad better.

File: libraries/dotnet/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+## Steps
+
+| Step | Description |
+| ----------- | ----------- |
+| `source_build` | This step leverages the dotnet publish command to build your application and output the results to the specified directory via outDir Variable. outDir defaults to a folder named "bin." The specified folder is archived as a jenkins artifact. |
```
Comment (ConnorDY) [inline libraries/dotnet/README.md:13]: Please add backticks (\`) around `dotnet publish` and `outDir` to make this styled a tad better. Also please capitalize the "J" in Jenkins.

File: libraries/dotnet/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+| `unit_test` | This step leverages the dotnet test command to run the unit, integration and functional tests specified in the application repository and outputs the results to a specified directory via resultDir variable. resultDir defaults to a folder named "coverage." The specified folder is archived as a jenkins artifact.|
```
Comment (ConnorDY) [inline libraries/dotnet/README.md:14]: Please add backticks (\`) around `dotnet test` and `resultDir` to make this styled a tad better. Also please capitalize the "J" in Jenkins.

File: libraries/dotnet/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+      resultDir = "Results"
+    }
+  }
+}
+```
+
+## Dependencies
+
+* The `sdp` library
+* Access to the dotnet-sdk:5.0 build agent container via the repository defined in your `sdp` library configuration
```
Comment (ConnorDY) [inline libraries/dotnet/README.md:34]: Please add backticks (\`) around `dotnet-sdk:5.0` to make this styled a tad better. I would remove them from around "sdp" and make SDP all caps.

File: libraries/dotnet/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+    unit_test {
+      resultDir = "Results"
+    }
+  }
+}
+```
+
+## Dependencies
+
+* The `sdp` library
```
Comment (ConnorDY) [inline libraries/dotnet/README.md:33]: Please remove the backticks (\`) around "sdp" and make SDP all caps.

File: libraries/dotnet/README.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
+---
+description: This library allows you to perform .Net build and test commands in the sdp dotnet-sdk:5.0 agent container
```
Comment (ConnorDY) [inline libraries/dotnet/README.md:2]: ".Net" --> ".NET"

**Review by ConnorDY — CHANGES_REQUESTED** [review]
Just a few very minor nitpicks...

**Comment by ConnorDY** [comment]
Oh, can you also update the PR title to be a short and well worded description of the PR changes.

For this PR, something like "Added null safe operators and documentation for the .NET library" could work!

**Review by psig-bah — APPROVED** [review]

**Review by ConnorDY — APPROVED** [review]

**Review by ConnorDY — APPROVED** [review]

**Pull request merged**